frontend/javascripts/admin/dataset/dataset_upload_view.js

```diff
@@ -336,6 +349,34 @@ class DatasetUploadView extends React.Component<PropsWithFormAndRouter, State> {
     }
   };
 
+  cancelUpload = async () => {
+    const { uploadId, resumableUpload, datasetId, datastoreUrl } = this.state;
+    resumableUpload.pause();
+    const shouldCancel = await confirmAsync({
+      title:
+        "Cancelling the running upload will delete all already uploaded files on the server and cannot be undone. Are you certain to cancel the upload?",
```

Comment:
```suggestion
        "Cancelling the running upload will delete all already uploaded files on the server and cannot be undone. Are you sure you want to cancel the upload?",
```

sounds more natural in my opinion